### PR TITLE
Emis

### PR DESCRIPTION
**Pull request by raydittmeier**

### CODE REVIEW

**Comment by gschmidt-usgs** [comment]
ST_DATA_DIR appears to be a new environment variable.  If so, that change should be noted in the README and documented as part of the installation instructions.

**Comment by gschmidt-usgs** [comment]
Peer review complete.

**Comment by raydittmeier** [comment]
ST_DATA_DIR is already used by some of the Surface Temperature scripts, so it's already documented in the "Environment Variables" section of the README.

File: provisional-st/scripts/estimate_landsat_emissivity.py [hunk 1]
```diff
@@ -876,10 +895,18 @@ def main():
         # Register all the gdal drivers
         gdal.AllRegister()
 
+        # Get the data directory from the environment
+        if 'ST_DATA_DIR' not in os.environ:
+            raise Exception('Environment variable ST_DATA_DIR is'
+                            ' not defined')
+        else:
+            st_data_dir = os.environ.get('ST_DATA_DIR')
+
```
Comment (ronaldddilley) [inline provisional-st/scripts/estimate_landsat_emissivity.py]: I have been doing this lately.  If you want to change it.

def get_env_var(variable, default):
    result = os.environ.get(variable, default)
    if result == None:
        raise RuntimeError('You must specify {} in the environment'
                           .format(variable))
    return result

File: provisional-st/scripts/estimate_landsat_emissivity_stdev.py [hunk 2]
```diff
@@ -215,6 +217,14 @@ def generate_tiles(src_info, url, wkt, no_data_value, intermediate):
     - Generate the Landsat EMIS standard deviation from the 13 and 14 stdev
     - band data
     '''
+
+    logger = logging.getLogger(__name__)
+
+    # Read the ASTER GED tile list
+    ged_tile_file = 'aster_ged_tile_list.txt'
+    with open(os.path.join(st_data_dir, ged_tile_file)) as ged_file:
+        tiles = [os.path.splitext(line.rstrip('\n'))[0] for line in ged_file]
```
Comment (ronaldddilley) [inline provisional-st/scripts/estimate_landsat_emissivity_stdev.py]: Same comment as above

File: provisional-st/scripts/estimate_landsat_emissivity_stdev.py [hunk 3]
```diff
@@ -239,9 +254,10 @@ def generate_tiles(src_info, url, wkt, no_data_value, intermediate):
                                 filename=filename,
                                 intermediate=intermediate))
 
-        # Skip the tile if it isn't available
+        # Fail if a tile can't be read, but it is in the ASTER GED
         if not aster_data_available:
-            continue
+            logger.error('Cannot reach tile {} in ASTER GED'.format(filename))
+            return (list())
```
Comment (ronaldddilley) [inline provisional-st/scripts/estimate_landsat_emissivity_stdev.py]: Same comment as above

File: provisional-st/scripts/estimate_landsat_emissivity_stdev.py [hunk 4]
```diff
@@ -466,10 +486,18 @@ def main():
         # Register all the gdal drivers
         gdal.AllRegister()
 
+        # Get the data directory from the environment
+        if 'ST_DATA_DIR' not in os.environ:
+            raise Exception('Environment variable ST_DATA_DIR is'
+                            ' not defined')
+        else:
+            st_data_dir = os.environ.get('ST_DATA_DIR')
+
```
Comment (ronaldddilley) [inline provisional-st/scripts/estimate_landsat_emissivity_stdev.py]: Same comment as above

File: provisional-st/scripts/estimate_landsat_emissivity.py [hunk 5]
```diff
@@ -452,9 +466,10 @@ def generate_tiles(src_info, coefficients, url, wkt,
                                 filename=filename,
                                 intermediate=intermediate))
 
-        # Skip the tile if it isn't available
+        # Fail if a tile can't be read, but it is in the ASTER GED
         if not aster_data_available:
-            continue
+            logger.error('Cannot reach tile {} in ASTER GED'.format(filename))
+            return (list(), list())
```
[the remark this answers is not in the document]
Comment (raydittmeier) [reply provisional-st/scripts/estimate_landsat_emissivity.py]: I'll raise an exception.

File: provisional-st/scripts/estimate_landsat_emissivity.py (hunk 1 above)
Comment (raydittmeier) [reply provisional-st/scripts/estimate_landsat_emissivity.py]: I'll use this code to build a function in emissivity_utilities.py and call it from the 2 emissivity tools.

**Pull request merged**